src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -159,6 +166,91 @@ internal void AddProvider(string workerDir)
             }
         }
 
+        private List<RpcWorkerDescriptionProfile> ReadWorkerDescriptionFromProfiles(JObject workerConfig)
+        {
+            List<RpcWorkerDescriptionProfile> descriptionProfiles = new List<RpcWorkerDescriptionProfile>();
+            JToken profiles = workerConfig.GetValue(RpcWorkerConstants.WorkerDescriptionProfiles);
+            if (profiles != null)
+            {
+                try
+                {
+                    descriptionProfiles = profiles.ToObject<List<RpcWorkerDescriptionProfile>>();
+                }
+                catch (Exception ex)
+                {
+                    _logger?.LogDebug(ex, $"Failed to parse profiles. Default worker description will be used.");
+                }
+            }
+            return descriptionProfiles;
+        }
+
+        private RpcWorkerDescription LoadWorkerDescriptionFromProfiles(List<RpcWorkerDescriptionProfile> descriptionProfiles, RpcWorkerDescription defaultWorkerDescription)
+        {
+            foreach (var profile in descriptionProfiles)
+            {
+                if ( profile.Validate() && CheckProfileConditions(profile))
+                {
+                    RpcWorkerDescription profileDescription = profile.ProfileDescription;
+                    profileDescription.Arguments = profileDescription.Arguments?.Count > 0 ? profileDescription.Arguments : defaultWorkerDescription.Arguments;
```

Comment:
Since `profileDescription.Arguments` should already be the default, I wonder if one of these might be cleaner to use:

a) single line if